### PR TITLE
Validate with trivy docker image

### PR DESCRIPTION
**Pull request by outscale-vbr**
**What type of PR is this?**
/kind feature  

**What this PR does / why we need it**:
Add trivy to scan image
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes  https://github.com/outscale-dev/cluster-api-provider-outscale/issues/145
https://github.com/outscale-dev/cluster-api-provider-outscale/issues/165

**Special notes for your reviewer**:
I fix also boilerplate template
_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Review by outscale-hmi — APPROVED** [review]
OK

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - 'main'
+    paths:
+      - "**.go"
+      - "**.yaml"
+      - "!capm.yaml"
+      - "!osc-secret.yaml"
+      - "!example/**.yaml"
+      - "!helm/**"
+      - "Makefile"
+      - "!docs/src/**"
```
Comment (outscale-mdr) [inline .github/workflows/docker.yaml:16]: this should also be in `pull_request` section

File: hack/check-trivy.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                        logger.debug("{} has been found but it is not resolved".format(cve))
+                    ignored_cves.remove(cve)
+                except KeyError:
+                    logger.warning("Status for {} has not been found".format(cve))
+    if resolved:
+        sys.exit(1)
+    elif len(ignored_cves) != 0:
+        logger.error("These CVE has not been found: {}".format(ignored_cves))
+        sys.exit(1)
+
```
Comment (outscale-mdr) [inline hack/check-trivy.py:90]: This was from osc-bsu-csi repo, it has not been updated yet after the migration to alpine. The use was to check that in debian if the CVE has been fixed or not.

Here you use alpine, so it does not work ad it is not useful. Maybe you could check how to update this script in order not to ignore fixed CVEs

File: hack/boilerplate/boilerplate.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
```
Comment (outscale-mdr) [inline hack/boilerplate/boilerplate.py:1]: unrelated files ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

**Comment by outscale-vbr** [comment]
Close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/145
Close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/165

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (outscale-vbr) [reply .github/workflows/docker.yaml]: Ok done (thanks)

File: hack/check-trivy.py (hunk 2 above)
Comment (outscale-vbr) [reply hack/check-trivy.py]: Good remark.
I rewrite the file to be compatible with alpine instead of debian.

File: hack/boilerplate/boilerplate.py (hunk 3 above)
Comment (outscale-vbr) [reply hack/boilerplate/boilerplate.py:1]: Boilerplate failed (so i include the fix also here and add link with the issue)

**Comment by outscale-vbr** [comment]
@outscale-mdr Are you ok with the changes ?

**Review by outscale-mdr — APPROVED** [review]
